lightning/src/ln/channelmanager.rs

```diff
@@ -3097,9 +3121,8 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 
 			for (short_chan_id, mut pending_forwards) in channel_state.forward_htlcs.drain() {
 				if short_chan_id != 0 {
-					let forward_chan_id = match channel_state.short_to_chan_info.get(&short_chan_id) {
-						Some((_cp_id, chan_id)) => chan_id.clone(),
-						None => {
+					macro_rules! fail_pending_forwards {
```

Comment:
Not super happy about this macro, and would prefer to break it out into a separate function due to the added complexity. But I figured `process_pending_htlc_forwards` was probably kept large with macros for efficacy reasons, so opted for the new macro instead. Let me know if my assumptions are wrong, and you'd prefer to break this out :)

Reply:
I'm not sure here, I just know that nested macros are not fun to debug :see_no_evil:

Reply:
Yeah definitely agree 😬...